mm2src/coins/hd_wallet/mod.rs

```diff
@@ -241,20 +242,61 @@ where
     }
 }
 
-pub async fn load_hd_accounts_from_storage<HDAddress, ExtendedPublicKey>(
+pub async fn load_hd_accounts_from_storage_with_matching_xpubs<HDAddress>(
```

Comment:
The new name is quite confusing at first glance to me because it doesn't take anything related to "xpubs" which is actually just an implementation detail of the function. I think instead of taking `global_hd_ctx` it should take `derived_xpub` directly. This would also make the function easier to test.

Reply:
[d5bfc97](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2482/commits/d5bfc9799f6fb40521ca80dccf86bff65fc1902a)